File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map <int, int> container;

        for(int i = 0; i < nums.size(); i++){
            container[target - nums[i]] = i;
        }
        vector<int> result;
        for(int i = 0; i < nums.size(); i++){
            if(container[nums[i]]){
                if(i == container[nums[i]]){
                    continue;
                }
                result.push_back(i);
                result.push_back(container[nums[i]]);
                return result;
            }
        }
        
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1 

        while numbers[left] + numbers[right] != target:
            add = numbers[left] + numbers[right]

            if add > target:
                right -= 1
            else:
                left += 1
        
        return [left + 1, right + 1]